datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,594 @@
+        writer: Tensorboard writer object
+        checkpointer: Model checkpointer callback to save models
+        device: model training on device (cpu|cuda)
+    """
+
+    def __init__(
+        self,
+        *,
+        config,
+        writer,
+        kfp_writer,
+        checkpointer,
+        device,
+        box_score_thresh=0.05,
+        gpu=0,
+        rank=0,
+        **kwargs,
+    ):
+        logger.info(f"initializing faster rcnn")
+        self.config = config
+        self.device = device
+        self.writer = writer
+        self.kfp_writer = kfp_writer
+        model_name = f"fasterrcnn_{self.config.backbone}_fpn"
+        self.model = torchvision.models.detection.__dict__[model_name](
+            num_classes=config.num_classes,
+            pretrained_backbone=config.pretrained_backbone,
+            pretrained=config.pretrained,
+            box_detections_per_img=MAX_BOXES_PER_IMAGE,
+            box_score_thresh=box_score_thresh,
+        )
+        self.model_without_ddp = self.model
+        self.gpu = gpu
+        self.rank = rank
+        self.sync_metrics = config.get("synchronize_metrics", True)
+        logger.info(f"gpu: {self.gpu}, rank: {self.rank}")
+
+        self.checkpointer = checkpointer
+        checkpoint_file = config.get("checkpoint_file", const.NULL_STRING)
+        if checkpoint_file != const.NULL_STRING:
+            checkpointer.load(self, config.checkpoint_file)
+
+        self.metrics = {}
+        for metric_key, metric in config.metrics.items():
+            self.metrics[metric_key] = EvaluationMetric.create(
+                metric.name, **metric.args
+            )
+        self.model.to(self.device)
+
+        if self.config.system.distributed:
+            self.model = torch.nn.parallel.DistributedDataParallel(
+                self.model, device_ids=[self.gpu]
+            )
+            self.model_without_ddp = self.model.module
+
+    def train(self, **kwargs):
+        config = self.config
+        train_dataset = download_data(config, TRAIN)
+        val_dataset = download_data(config, VAL)
+        label_mappings = train_dataset.label_mappings
+        if self.config.system.dryrun:
+            train_dataset = create_dryrun_dataset(config, train_dataset, TRAIN)
+            val_dataset = create_dryrun_dataset(config, val_dataset, VAL)
+
+        logger.info(f"length of train dataset is {len(train_dataset)}")
+        logger.info(f"length of validation dataset is {len(val_dataset)}")
+        is_distributed = config.system.distributed
+        train_sampler = create_sampler(is_distributed=is_distributed,
+                                       dataset=train_dataset,
+                                       is_train=True)
+        val_sampler = create_sampler(is_distributed=is_distributed,
+                                     dataset=val_dataset,
+                                     is_train=False)
+
+        train_loader = create_dataloader(config, train_dataset, train_sampler,
```

Comment:
Why is create_sampler a module level method? Can this be staticmethod for the FasterRCNN class?

Reply:
yes